src/util/instanceEdit.tsx

```diff
@@ -120,8 +120,23 @@ export const InstanceEditSchema: Yup.ObjectSchema<{
   instanceType: Yup.string().required("Instance type is required"),
 });
 
-export const ensureEditMode = (formik: ConfigurationRowFormikProps) => {
+export const ensureEditMode = (
+  formik: ConfigurationRowFormikProps,
+  fieldName: string,
+) => {
   if (formik.values.readOnly) {
     void formik.setFieldValue("readOnly", false);
   }
+  trackChangedField(formik, fieldName);
+};
+
+export const trackChangedField = (
+  formik: ConfigurationRowFormikProps,
+  fieldName?: string,
+) => {
+  const oldValues = formik.values.changedFields ?? new Set<string>();
+  const changedFields = new Set([...oldValues, fieldName]);
+  if (changedFields.size > oldValues.size) {
+    void formik.setFieldValue("changedFields", changedFields);
+  }
 };
```

Comment:
Here is an rough idea (excuse the messy code), maybe we can do something like the below example to track field value changes against initial form values? There might be issues with the comparison since here I am assuming that field values have primitive types always e.g. if a field value is an object, then we would probably need a deep comparison.

Lastly, we probaly don't have to call this function in `ensureEditMode` and `enableOverride`.

```suggestion
export const trackChangedField = (
  formik: ConfigurationRowFormikProps,
  fieldName?: string,
) => {
  const oldValues = formik.values.changedFields ?? new Set<string>();
  const changedFields = new Set(oldValues);

  const values = formik.values as unknown as Record<string, string | undefined>;
  const initialValues = formik.initialValues as unknown as Record<
    string,
    string | undefined
  >;

  if (fieldName) {
    if (values[fieldName] === initialValues[fieldName]) {
      changedFields.delete(fieldName);
    } else {
      changedFields.add(fieldName);
    }
  }

  if (changedFields.size !== oldValues.size) {
    void formik.setFieldValue("changedFields", changedFields);
  }
};
```

Reply:
Using initial values is a good idea. That works well on primitive fields.
As you mention: unsolved are changes to devices and attached profiles, both are complex fields which are important to get right as well.

Reply:
Using initial values is a great idea. I completely changed the approach now, because we don't need to track the lifecycle of changes, we can calculate the changes on the fly based on the initial values and comparing it to current values. It is a bit involved to get that to a single number for complex objects, but much easier on containing the complexity.